### PR TITLE
Fix cryptiles vulnerability

### PR DESCRIPTION
**Pull request by akanduru**

### CODE REVIEW

File: src/adapters/node.js [hunk 1]
```diff
@@ -8,11 +7,20 @@
         defer: Q.defer,
         http: function(args) {
             var deff = Q.defer();
-            args.body = args.data;
+            if(args.data && typeof args.data === "string") {
+              // Setting args.json requires args.body to be a JSON as per request docs.
+              args.body = JSON.parse(args.data);
```
Comment (mike1pol) [inline src/adapters/node.js]: Need need to use construction try catch for JSON.parse, string may is not JSON format

Comment (akanduru) [reply src/adapters/node.js]: Hmm, I would rather through parsing error to the caller than proceed further with the request. args.json is set to true. We are saying the input should be json...

Comment (mike1pol) [reply src/adapters/node.js]: You can catch it and detailed describe.

Comment (akanduru) [reply src/adapters/node.js]: @mike1pol: I just realized that we are following the wrong thread. The PR #127 supersedes this. It is better to discuss all these issues there. If you agree I will close this PR?

Comment (mike1pol) [reply src/adapters/node.js]: Ok

**Pull request closed without merging**